node/metrics.go

```diff
@@ -225,12 +228,29 @@ func (g *Metrics) collectOnchainMetrics() {
 			})
 			for i, op := range operatorStakeShares {
 				if op.operatorId == g.operatorId {
+					g.allQuorumCache[q] = true
 					g.RegisteredQuorumsStakeShare.WithLabelValues(fmt.Sprintf("%d", q)).Set(op.stakeShare)
 					g.RegisteredQuorumsRank.WithLabelValues(fmt.Sprintf("%d", q)).Set(float64(i + 1))
 					g.logger.Info("Current operator registration onchain", "operatorId", g.operatorId.Hex(), "blockNumber", blockNum, "quorumId", q, "stakeShare (basis point)", op.stakeShare, "rank", i+1)
 					break
 				}
 			}
 		}
+		// Check if operator deregistered for an existing quorum, set the stake share and rank to 0
+		g.ResetQuorumMetrics(blockNum)
+	}
+}
+
+func (g *Metrics) ResetQuorumMetrics(blockNum uint32) {
+	// Check if operator deregistered for an existing quorum, set the stake share and rank to 0
+	for q := range g.allQuorumCache {
+		// If this quorum was deregistered then set the stake share and rank to 0
+		if !g.allQuorumCache[q] {
```

Comment:
So if a quorum is deregistered, for each cycle, it will continue to set the value to 0, is that expected? Or you just need to set it to 0 once (then it'll keep using that latest value if no newer value)?

Reply:
It's fine to keep setting 0, similar to if the stake doesn't change it still sets it to previous value. Doesn't matter much IMO. checking that what was the previous value and update based on just adds more complexity to code so we can keep it simple